gfx/include/public/gfx/gfx_typedefs.h

```diff
@@ -0,0 +1,12 @@
+// Copyright Antioch. All rights reserved.
+
+#pragma once
+
+#include <cstddef>
+
+namespace antioch::gfx {
+
+typedef void* (*PFN_allocateFunction)(void* pUserData, size_t size);
```

Comment:
do you really need these?

Reply:
in the case where the allocation we are using is different from just new (i.e. allocate from pool) it could be useful